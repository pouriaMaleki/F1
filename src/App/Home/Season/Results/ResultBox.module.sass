
.root
  font-family: Kanit-ExtraBold
  width: 260px
  height: 20px
  line-height: 20px
  background-color: #222
  padding: 20px
  vertical-align: top

.title
  display: inline-block
  float: left

.value
  display: inline-block
  float: right
  color: #ff2000
