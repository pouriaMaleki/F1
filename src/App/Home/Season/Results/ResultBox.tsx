import React from "react";
import css from "./ResultBox.module.sass";
import classNames from "classnames";
import { IResult } from "../IResult";

interface IProps extends IResult {
  className?: string;
}

const ResultBox = (props: IProps) => (
  <div className={classNames(css.root, props.className)}>
    <div className={css.title}>{props.title}</div>
    <div className={css.value}>{props.value}</div>
  </div>
);

export { ResultBox };
