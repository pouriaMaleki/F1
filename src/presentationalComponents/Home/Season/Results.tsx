import React from "react";
import css from "./Results.module.sass";
import classNames from "classnames";
import { ResultBox } from "./Results/ResultBox";
import { IResult } from "./IResult";

interface IProps {
  data: IResult[];
  className?: string;
}

const Results = (props: IProps) => (
  <div className={classNames(css.root, props.className)}>
    {props.data.map((result, index) => (
      <ResultBox key={index + ""} {...result} />
    ))}
  </div>
);

export { Results };
