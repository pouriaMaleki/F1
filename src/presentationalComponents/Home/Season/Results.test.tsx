import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { Results } from "./Results";
import { ResultBox } from "./Results/ResultBox";

const data = [
  {
    title: "No.",
    value: "26"
  },
  {
    title: "No.",
    value: "26"
  },
  {
    title: "No.",
    value: "26"
  }
];

describe("Render Results", () => {
  // snapshot test
  it("render correctly Results component", () => {
    const ResultsComponent = renderer.create(<Results data={data} />).toJSON();
    expect(ResultsComponent).toMatchSnapshot();
  });

  // test props
  it("add className as props", () => {
    const CLASSNAME = "CLASSNAME";
    const props = {
        data: [],
        className: CLASSNAME
      },
      ResultsComponent = mount(<Results {...props} />);

    expect(ResultsComponent.hasClass(CLASSNAME)).toEqual(true);
  });

  it("render all results", () => {
    const props = {
        data
      },
      ResultsComponent = mount(<Results {...props} />);

    expect(ResultsComponent.find(ResultBox).length).toEqual(data.length);
  });
});
