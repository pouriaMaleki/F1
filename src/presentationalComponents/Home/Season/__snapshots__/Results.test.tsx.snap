// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Results render correctly Results component 1`] = `
<div
  className="root"
>
  <div
    className="root"
  >
    <div
      className="title"
    >
      No.
    </div>
    <div
      className="value"
    >
      26
    </div>
  </div>
  <div
    className="root"
  >
    <div
      className="title"
    >
      No.
    </div>
    <div
      className="value"
    >
      26
    </div>
  </div>
  <div
    className="root"
  >
    <div
      className="title"
    >
      No.
    </div>
    <div
      className="value"
    >
      26
    </div>
  </div>
</div>
`;
